components/sup/src/config.rs

```diff
@@ -158,7 +155,12 @@ impl FromStr for GossipListenAddr {
             Ok(addr) => Ok(GossipListenAddr(addr)),
             Err(_) => match IpAddr::from_str(val) {
                 Ok(ip) => {
-                    let mut addr = GossipListenAddr::default();
+                    // We shouldn't need to fallback to default here. The value
+                    // should always be set at the clap layer. I'm leaving this
+                    // here for now so as to not break anything unexpected, but
+                    // we should revisit this once we have the default bits all
+                    // moved to the clap layer.
+                    let mut addr = GossipListenAddr::configured_value();
```

Comment:
Whoa, this is weird... I didn't realize we were using default down here in `FromStr`. Now that we've got this hooked up to `EnvConfig`, it creates a very confusing cyclic dependency (`EnvConfig` is itself implemented in terms of `FromStr`, but _this_ `FromStr` is now implemented in terms of `EnvConfig`!)

I think it's worth tracing this through to see where all this is being called. It's also odd that even though the contract of the `from_str` method is to return a `Result`, because of the `Default` fallback behavior, we're never going to return an error.

It seems like the original implementation wanted to be something like `EnvConfig`, but without being declared as  formally. I think we have the ability to clean that up now.